pkg/expression/explain.go

```diff
@@ -36,6 +37,14 @@ func (expr *ScalarFunction) explainInfo(ctx EvalContext, normalized bool) string
 	intest.Assert(normalized || ctx != nil)
 	var buffer bytes.Buffer
 	fmt.Fprintf(&buffer, "%s(", expr.FuncName.L)
+	// convert `in(_tidb_pid, -1)` to `in(_tidb_pid, dual)` whether normalized equals to true or false.
+	if expr.FuncName.L == ast.In {
+		args := expr.GetArgs()
+		if len(args) == 2 && args[0].ExplainNormalizedInfo() == model.ExtraPartitionIdName.L && args[1].(*Constant).Value.GetInt64() == -1 {
```

Comment:
@tiancaiamao should we add this? Looks a little bit weird.

Reply:
In `selection`, normal partition IDs are also numbers, and -1 feels fine for me.